specs/preinstalls.md

```diff
@@ -0,0 +1,206 @@
+
+## Safe
+
+[Implementation](https://github.com/safe-global/safe-contracts/blob/v1.3.0/contracts/GnosisSafe.sol)
+
+Address: `0x69f4D1788e39c87893C980c06EdF4b7f686e2938`
+
+A multisignature wallet with support for confirmations using signed messages based on ERC191.
+Differs from [SafeL2](#safel2) by not emitting events to save gas.
+
+## SafeL2
+
+[Implementation](https://github.com/safe-global/safe-contracts/blob/v1.3.0/contracts/GnosisSafeL2.sol)
+
+Address: `0xfb1bffC9d739B8D520DaF37dF666da4C687191EA`
+
+A multisignature wallet with support for confirmations using signed messages based on ERC191.
+Differs from [Safe](#safe) by emitting events.
+
+## MultiSend
+
+[Implementation](https://github.com/safe-global/safe-contracts/blob/v1.3.0/contracts/libraries/MultiSend.sol)
+
+Address: `0x998739BFdAAdde7C933B942a68053933098f9EDa`
+
+Allows to batch multiple transactions into one.
+
+## MultiSendCallOnly
+
+[Implementation](https://github.com/safe-global/safe-contracts/blob/v1.3.0/contracts/libraries/MultiSendCallOnly.sol)
+
+Address: `0xA1dabEF33b3B82c7814B6D82A79e50F4AC44102B`
+
+Allows to batch multiple transactions into one, but only calls.
+
+## SafeSingletonFactory
+
+[Implementation](https://github.com/safe-global/safe-singleton-factory/blob/main/source/deterministic-deployment-proxy.yul)
+
+Address: `0x914d7Fec6aaC8cd542e72Bca78B30650d45643d7`
+
+Singleton factory used by Safe-related contracts based on
+[Arachnid's Deterministic Deployment Proxy](#arachnids-deterministic-deployment-proxy).
+
+The original library used a pre-signed transaction without a chain ID to allow deployment on different chains.
+Some chains do not allow such transactions to be submitted; therefore, this contract will provide the same factory
+that can be deployed via a pre-signed transaction that includes the chain ID. The key that is used to sign is
+controlled by the Safe team.
+
+## MultiCall3
```

Comment:
big nit, usually stylized with a lowercase "c" (other places to change this below too)
```suggestion
## Multicall3
```

Reply:
@mds1 Oof I've used `C` all over the code and docs - is it okay just to change it here, should we leave it so it matches all other mentions?

Reply:
code is internal, so I think just changing it in docs here for consistency with the contract itself and [other docs](https://viem.sh/docs/clients/public.html#eth_call-aggregation-via-multicall) is sufficient